### ISSUE TITLE
gonative fails to build

### ISSUE THREAD

**Issue by tchap** [inconshreveable/gonative#20]
```
$ godep go install
# github.com/inconshreveable/axiom
../axiom/axiom.go:108: too few values in struct initializer
../axiom/axiom.go:109: too few values in struct initializer
../axiom/axiom.go:110: too few values in struct initializer
../axiom/axiom.go:231: too few values in struct initializer
../axiom/axiom.go:330: too few values in struct initializer
godep: go exit status 2
$ go version
go version go1.5.1 darwin/amd64
$ godep version
godep v3 (darwin/amd64/go1.5)
```

**Comment by tchap** [comment]
Hmm, I guess that it should be possible to use Godep, I guess the issue is that regular `go get` won't work... Somehow don't manage to get that to work either...

**Comment by tchap** [comment]
Well, after creating a clean workspace:
```bash
$ go get -d github.com/inconshreveable/gonative
$ cd src/github.com/inconshreveable/gonative
$ make
go get github.com/tools/godep
WORKSPACE/src/github.com/inconshreveable/gonative/Godeps/_workspace/bin/godep restore
godep: restore, during download dep: exit status 1
make: *** [workspace] Error 1
```

**Comment by geoko86** [comment]
I am running into the same issue. Using Mac OS X with a clean workspace. When I run 
```
godep restore -v 
```
I get these errors:
```
execute: &{Path:/usr/local/go/bin/go Args:[go get -d -v bitbucket.org/kardianos/osext] Env:[] Dir:. Stdin:<nil> Stdout:bitbucket.org/kardianos/osext (download)
package bitbucket.org/kardianos/osext: code in directory /Users/gkoester/development/golang_temp/src/bitbucket.org/kardianos/osext expects import "github.com/kardianos/osext"
 Stderr:bitbucket.org/kardianos/osext (download)
package bitbucket.org/kardianos/osext: code in directory /Users/gkoester/development/golang_temp/src/bitbucket.org/kardianos/osext expects import "github.com/kardianos/osext"
 ExtraFiles:[] SysProcAttr:<nil> Process:0xc820104040 ProcessState:exit status 1 lookPathErr:<nil> finished:true childFiles:[0xc820030080 0xc820030098 0xc820030098] closeAfterStart:[0xc820030080 0xc820030098] closeAfterWait:[0xc820030090] goroutine:[0x9b6b0] errch:0xc820018660}
 output: bitbucket.org/kardianos/osext (download)
package bitbucket.org/kardianos/osext: code in directory /Users/gkoester/development/golang_temp/src/bitbucket.org/kardianos/osext expects import "github.com/kardianos/osext"
```

The issue seems to be in the osext package from bitbucket.
```
bitbucket.org/kardianos/osext expects import "github.com/kardianos/osext"
```

Does anybody have an idea?

**Comment by tchap** [comment]
This actually only completely breaks when using Go 1.5 I think. I solved the issue for myself right now by installing `gvm` and using Go 1.4.3. In this case `godep restore -v` still breaks on the first run, but when you run it again, it somehow succeeds and then `godep go install` works fine. So my script basically does this:

1. Install `gvm`.
2. `gvm install go1.4.3`
3. `gvm use go1.4.3`
4. Clone gonative.
5. Run `godep restore -v` 3 times. In case we always get `$? -ne 0`, we abort.
6. `godep go install` works now.

**Comment by tchap** [comment]
The solution could be I guess to update the version of `github.com/kardianos/osext` to a more recent commit. Will take a look...

**Comment by tchap** [comment]
Ah, it's still importing the old import path, that is why it is breaking. The right import path is `github.com/kardianos/osext`, obviously...

**Comment by geoko86** [comment]
true, I am using Go 1.5.1. After fixing the Godeps.json by Hand i was able to run make again:
```
{
        "ImportPath": "_/Users/aes/src/gonative",
        "GoVersion": "go1.3",
        "Packages": [
                "github.com/inconshreveable/gonative"
        ],
        "Deps": [
                {
                        "ImportPath": "github.com/kardianos/osext",
                        "Comment": "null-15",
                        "Rev": "10da29423eb9a6269092eebdc2be32209612d9d2"
                },
                {
                        "ImportPath": "github.com/codegangsta/cli",
                        "Comment": "1.2.0-50-ga14c5b4",
                        "Rev": "a14c5b47c7efa4ff80cc42e1079a34b4756f2311"
                },
                {
                        "ImportPath": "github.com/inconshreveable/axiom",
                        "Rev": "d47f34fee0ce318eb07f0eee4b0e68b5662355ed"
                },
                {
                        "ImportPath": "github.com/inconshreveable/go-update",
                        "Rev": "8455de157e5e3eee5d680febf627fc4c03b59332"
                },
                {
                        "ImportPath": "github.com/inconshreveable/mousetrap",
                        "Rev": "76626ae9c91c4f2a10f34cad8ce83ea42c93bb75"
                },
                {
                        "ImportPath": "github.com/kr/binarydist",
                        "Rev": "9955b0ab8708602d411341e55fffd7e0700f86bd"
                },
                {
                        "ImportPath": "github.com/mattn/go-colorable",
                        "Rev": "043ae16291351db8465272edf465c9f388161627"
                },
                {
                        "ImportPath": "gopkg.in/inconshreveable/log15.v2",
                        "Comment": "v2.7",
                        "Rev": "7cf5571b3b32a7c7b5605dc5ed8ad8fdbb2135f0"
                },
                {
                        "ImportPath": "gopkg.in/yaml.v1",
                        "Rev": "9f9df34309c04878acc86042b16630b0f696e1de"
                }
        ]
}
```

**Comment by geoko86** [comment]
Had to update ImportPath and Rev for "osext" and had to update the Rev for "go-update" and "axiom"

**Comment by tchap** [comment]
You can send a pull request ;-)

**Comment by geoko86** [comment]
send pull request https://github.com/inconshreveable/gonative/pull/22

**Comment by tchap** [comment]
Cheers. Now we only need to wait for someone to merge it :-) @inconshreveable

**Comment by inconshreveable** [comment]
Fixed by #23

**Comment by inconshreveable** [comment]
Thanks guys!
